scripts/lib/CIME/case/check_input_data.py

```diff
@@ -205,13 +279,80 @@ def check_input_data(case, protocol="svn", address=None, input_data_root=None, d
+                                if chksum_file and firstdownload:
+                                    # Get the md5 checksum file
+                                    got_chksum = _download_checksum_file(server, input_data_root, chksum_file, local_user)
+                                    firstdownload = False
+                                isdirectory=rel_path.endswith(os.sep)
+                                no_files_missing = _download_if_in_repo(server,
+                                                                        input_data_root, rel_path.strip(os.sep),
+                                                                        local_user, isdirectory=isdirectory)
+                                if got_chksum and no_files_missing and not isdirectory:
+                                    verify_chksum(input_data_root,chksum_file, rel_path.strip(os.sep))
                         else:
+                            if chksum:
+                                verify_chksum(input_data_root,chksum_file, rel_path.strip(os.sep))
+                                logger.info("Chksum passed for file {}".format(os.path.join(input_data_root,rel_path)))
                             logging.debug("  Already had input file: '{}'".format(full_path))
-
                 else:
                     model = os.path.basename(data_list_file).split('.')[0]
                     logging.warning("Model {} no file specified for {}".format(model, description))
 
     return no_files_missing
+
+def _check_permissions_and_lock_inputdata_file(file_path, user):
+    basedir = os.path.dirname(file_path)
+    if not os.path.exists(basedir):
+        try:
+            os.makedirs(basedir)
+        except IOError as x:
+            expect(False, "Write to directory {} failed with error {}: {}".
+                   format(basedir, x.errno, x.strerror))
+
+    # check for existing lock
+    lock_file = file_path+".LOCK"
+    # if the file exists print it's contents and error out
+    if os.path.isfile(lock_file):
+        with open(lock_file) as f:
+            line = f.readline()
+        expect(False,line)
+    # if it doesn't exist create it
+    with open(lock_file, "w") as f:
+        f.write("File {} locked by user {} on {} at {}".
+                format(file_path, user,
+                       time.strftime("%y%m%d"), time.strftime("%H%M%S")))
+
+    return lock_file
+
+def verify_chksum(input_data_root, checksumfile, filename):
+    if not chksum_hash:
+        hashfile = os.path.join(input_data_root, local_chksum_file)
+        if not os.path.isfile(hashfile):
+            expect(False, "Failed to find or download file {}".format(hashfile))
+
+        with open(hashfile) as fd:
+            lines = fd.readlines()
+            for line in lines:
+                fchksum, fname = line.split()
+                if fname in chksum_hash:
+                    expect(chksum_hash[fname] == fchksum, " Inconsistant hashes in chksum for file {}".format(fname))
```

Comment:
This is a good error-check - thanks.

Minor spelling error: Inconsistant should be Inconsistent.